infra/docker/api/prod/Dockerfile

```diff
@@ -1,69 +1,73 @@
+FROM pruned as installer
 
-COPY apps/hash-api/codegen.config.ts apps/hash-api/codegen.config.ts
-COPY apps/hash-api/src/graphql/type-defs apps/hash-api/src/graphql/type-defs
-COPY libs/@local/hash-graphql-shared/src/graphql libs/@local/hash-graphql-shared/src/graphql
-COPY libs/@local/hash-graphql-shared/src/queries libs/@local/hash-graphql-shared/src/queries
+WORKDIR /usr/local/src/hash
 
-RUN yarn workspace @apps/hash-api codegen
+COPY --from=pruned /app/out/json/ .
+COPY --from=pruned /app/out/yarn.lock ./yarn.lock
 
-COPY libs/@local/hash-graphql-shared/codegen.config.ts libs/@local/hash-graphql-shared/codegen.config.ts
+RUN yarn install --frozen-lockfile --prefer-offline \
+    && yarn cache clean
 
-RUN yarn workspace @local/hash-graphql-shared codegen
+COPY --from=pruned /app/out/full/ .
 
-COPY libs/@local/hash-isomorphic-utils/codegen.config.ts libs/@local/hash-isomorphic-utils/codegen.config.ts
+ENV PYTHONUNBUFFERED=1 \
+    PYTHONDONTWRITEBYTECODE=1 \
+    PIP_DISABLE_PIP_VERSION_CHECK=on \
+    POETRY_ACTIVE=1 \
+    POETRY_HOME="/opt/poetry" \
+    POETRY_VIRTUALENVS_IN_PROJECT=true \
+    POETRY_NO_INTERACTION=1 \
+    POETRY_VERSION=1.4.2
+ENV VIRTUAL_ENV=".venv"
+ENV PATH="$POETRY_HOME/bin:$VIRTUAL_ENV/bin:$PATH"
 
-RUN yarn workspace @local/hash-isomorphic-utils codegen
+RUN apt-get update && \
+    apt-get install -y --no-install-recommends build-essential libffi-dev curl default-jre
 
-COPY apps/hash-api apps/hash-api
-COPY libs/@local/eslint-config libs/@local/eslint-config
-COPY libs/@local/hash-backend-utils libs/@local/hash-backend-utils
-COPY libs/@local/hash-graph-client libs/@local/hash-graph-client
-COPY libs/@local/hash-graphql-shared libs/@local/hash-graphql-shared
-COPY libs/@local/hash-isomorphic-utils libs/@local/hash-isomorphic-utils
-COPY libs/@local/hash-subgraph libs/@local/hash-subgraph
-COPY libs/@local/advanced-types libs/@local/advanced-types
-COPY libs/@local/tsconfig libs/@local/tsconfig
-COPY libs/@local/status libs/@local/status
-RUN mkdir -p /app/var/uploads
+# `apt-get install python3` installs an older version of Python and we want 3.11, copying it from the official image
+#   seems to work with just these commands, so we opt to do this rather than needing to build it from source or
+#   install it from an external source such as https://github.com/indygreg/python-build-standalone (see frontend vercel-install)
+#   if this causes problem, or we update dependencies, we can change approach.
 
+COPY --from=python-base /usr/local/bin/python3 /usr/local/bin/python3
+COPY --from=python-base /usr/local/bin/python3.11 /usr/local/bin/python3.11
+COPY --from=python-base /usr/local/lib/python3.11 /usr/local/lib/python3.11
+COPY --from=python-base /usr/local/lib/libpython3.so /usr/local/lib/libpython3.so
+COPY --from=python-base /usr/local/lib/libpython3.11.so /usr/local/lib/libpython3.11.so
+COPY --from=python-base /usr/local/lib/libpython3.11.so.1.0 /usr/local/lib/libpython3.11.so.1.0
```

Comment:
Not too happy that we had to copy this so many times. Perhaps there's a way we can make this more extensible when we need python for the turbo build graph?

Reply:
The python3 in bullseye is not 3.11